packages/groqd/src/commands/projectField.test.ts

```diff
@@ -109,6 +114,20 @@ describe("field (naked projections)", () => {
 
       expectTypeOf<ResultType>().toEqualTypeOf<Array<ImagesArray | null>>();
     });
+
+    it("can correctly project deeply optional values", () => {
+      type DeepOptional = {
+        image?: {
+          asset?: {
+            _type: "reference";
+          };
+        };
+      };
+      const query = q.as<DeepOptional>().field("image.asset");
```

Comment:
@soleoak Here is where I added a specific unit test for the reported issue.  It failed at first, obviously, and now it verifies that the original problem is solved 🎉